ds701_book/26-TimeSeries.qmd

```diff
@@ -0,0 +1,1351 @@
+plt.plot(forecast_index, forecast_values, label='Forecast', color='red')
+plt.fill_between(forecast_index, confidence_intervals.iloc[:, 0], confidence_intervals.iloc[:, 1], color='pink', alpha=0.3)
+plt.legend()
+plt.show()
+```
+
+The light pink area shows the 95% confidence interval for the forecast.
+
+**Questions**: what observations do you have about the prediction?
+
+
+# Model Evaluation and Forecasting
+
+## Criteria for model selection
+
+- **AIC (Akaike Information Criterion)**: A measure of the relative quality of a
+  statistical model for a given set of data. It is defined as:
+
+$$
+\text{AIC} = 2k - 2\ln(L)
+$$
+
+where:
+
+- $k$ is the number of parameters in the model,
+- $L$ is the maximum value of the likelihood function for the model.
+
+**BIC (Bayesian Information Criterion)**: Similar to AIC but includes a penalty term for the number of parameters in the model. It is defined as:
+
+$$
+\text{BIC} = k\ln(n) - 2\ln(L)
+$$
+
+where:
+
+- $k$ is the number of parameters in the model,
+- $n$ is the number of data points,
+- $L$ is the maximum value of the likelihood function for the model.
+
+**Key Points**:
+
+1. **Model Comparison**: Both AIC and BIC are used to compare different models; the model with the lower AIC or BIC is preferred.
+2. **Penalty for Complexity**: BIC imposes a larger penalty for models with more parameters compared to AIC.
+3. **Trade-off**: There is a trade-off between goodness of fit and model complexity.
+
+## Cross-validation techniques for time series
+
+- **Time Series Cross-Validation**: Unlike traditional cross-validation, time series cross-validation respects the temporal order of the data. 
+
+Common techniques include:
+
+- **Rolling Forecast Origin**: The training set is expanded with each iteration, and the model is re-evaluated.
+- **Time Series Split**: The data is split into multiple training and test sets, ensuring that the training set always precedes the test set.
+- **Blocked Cross-Validation**: The data is divided into blocks, and each block is used as a test set while the preceding blocks are used for training.
+
+## Key Points:
+1. **Respect Temporal Order**: Ensure that the training set always precedes the test set to avoid data leakage.
+2. **Multiple Techniques**: Various techniques like rolling forecast origin, time series split, and blocked cross-validation can be used.
+
+## Forecasting and confidence intervals
+
+- **Forecasting**: The process of making predictions about future values based on historical data. Common methods include:
+  - **ARIMA (AutoRegressive Integrated Moving Average)**: A popular time series forecasting method that combines autoregression, differencing, and moving average components.
+  - **Exponential Smoothing**: A technique that applies decreasing weights to past observations, giving more importance to recent data.
+  - **Prophet**: A forecasting tool developed by Facebook that handles seasonality, holidays, and missing data.
```

Comment:
I think we should drop this. It seems too random to throw it in there. We could mention it as a comment that there are other forecasting tools that are outside the scope of this lecture.

Reply:
ok. drop which? Prophet?

Reply:
dropped